server/impl/src/main/java/com/walmartlabs/concord/server/process/queue/dispatcher/Dispatcher.java

```diff
@@ -211,28 +213,37 @@ private List<Match> match(DSLContext tx, List<Request> requests) {
         return matches;
     }
 
-    private static Request findRequest(ProcessQueueEntry candidate, List<Request> requests) {
-        for (Request req : requests) {
-            Map<String, Object> capabilities = req.request.getCapabilities();
-            Map<String, Object> m = getAgentRequirements(candidate);
-            if (m.isEmpty() || Matcher.matches(capabilities, m)) {
-                return req;
-            }
+    static Request findRequest(ProcessQueueEntry candidate, List<Request> requests) {
+        return requests.stream()
+                .filter(req -> isRequestMatch(candidate, req))
+                .findFirst()
+                .orElse(null);
+    }
+
+    private static boolean isRequestMatch(ProcessQueueEntry candidate, Request req) {
+        try {
+            var capabilities = req.request.getCapabilities();
+            var requirements = getAgentRequirements(candidate);
+
+            return requirements.isEmpty() || Matcher.matches(capabilities, requirements);
+        } catch (PatternSyntaxException pse) {
+            log.error("Invalid regex in requested agent capabilities for instanceId: {}", candidate.key().getInstanceId());
```

Comment:
Maybe we should have a "shut up" command that we can send to agents with bad configuration. :-)
I.e. force those agents into maintenance mode.

Reply:
It's the reverse. Agent configs are fine in this case. However, If a `concord.yml` has a bad `configuration.requirements.agent.whatever` regex pattern, then it causes the dispatcher to explode and fail to dispatch any more processes.

Reply:
I do think some more defense against letting garbage in is better, but it's a little more of an open landscape to pre-validate (compile) all fields as a regular expression.

Reply:
I think further refinements here could be in the direction of kicking the nasty process request out so it won't continually be re-processed. Something like one of:
* Set the process to `FAILED` or `CANCELLED`. Fix your junk and try again.
* Assume no requirements if the provided value(s) cannot be read. You get dispatched to a default agent. *May* have side effects.

Reply:
I feel like marking as FAILED is an appropriate reaction. With some amount of cursing in the process log :-)